vcd/resource_vcd_vapp_vm.go

```diff
@@ -312,6 +315,32 @@ func resourceVcdVAppVmUpdate(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error getting VM status: %#v", err)
 	}
 
+	if d.HasChange("metadata") {
```

Comment:
I'd think if there is a change to improve this update only delete what was present and create what was missing rather than flushing all and adding all (and I know this was so in vApp).
This should in theory be better performing (less api calls) and would avoid temporary removal of tags which are not meant to be removed in the end.
What are your thoughts about this?

Reply:
I can implement that.

Reply:
Would be nice!

Reply:
Fixed.

Reply:
Hi, thanks! Only I can't find a commit for the recent fixes you wrote about.

Reply:
True because I did fixup the commit to my old commit. I like rebasing more.

Reply:
Got it. Please push separate commits for fixes next time - it's a lot easier to review the changes in such way. We will squash & merge at the end either way. Of course, we'll ask for a thorough commit message to accompany the single commit.